src/Components/SmartComponents/CVEs/CVEsTableToolbar.js

```diff
@@ -36,42 +36,41 @@ import { fetchCvesIds } from '../../../Store/Actions/Actions';
 import { setCvesWithoutErrata } from '../../../Helpers/APIHelper';
 import { getCveDefaultFilters } from './CVEsAssets';
 import { useHybridSystemFilterFlag } from '../../../Helpers/Hooks';
+import { AccountStatContext } from '../../../Utilities/VulnerabilityRoutes';
 
 const CVEsTableToolbarWithContext = ({ context, canEditStatusOrBusinessRisk, canExport, canToggleCvesWithoutErrata, intl }) => {
     const [exportPDF, setExportPDF] = useState(false);
 
     const { cves, params, methods, selectedCves } = context;
-    const { isLoading } = cves;
 
     const { filter } = params;
     const selectedCvesCount = selectedCves && selectedCves.length;
 
     const [showCvesWithoutErrata, setShowCvesWithoutErrata] = useState(null);
     const shouldUseHybridSystemFilter = useHybridSystemFilterFlag();
+    const { includesCvesWithoutErrata } = useContext(AccountStatContext);
 
     useEffect(() => {
-        if (!isLoading) {
-            // if the API response feature flag value differs currently saved one
-            if (showCvesWithoutErrata !== cves?.meta?.cves_without_errata) {
-                if (cves?.meta?.cves_without_errata === true) {
-                    if (!params.advisory_available) {
-                        methods.apply({ advisory_available: 'true' });
-                    }
+        // if the API response feature flag value differs currently saved one
+        if (showCvesWithoutErrata !== includesCvesWithoutErrata) {
+            if (includesCvesWithoutErrata === true) {
+                if (!params.advisory_available) {
+                    methods.apply({ advisory_available: 'true' });
                 }
-                else {
-                    if (params.advisory_available) {
-                        methods.apply({ advisory_available: undefined });
-                    }
+            }
+            else {
+                if (params.advisory_available) {
+                    methods.apply({ advisory_available: undefined });
                 }
             }
-
-            setShowCvesWithoutErrata(cves?.meta?.cves_without_errata);
         }
-    }, [cves?.meta?.cves_without_errata, isLoading]);
+
+        setShowCvesWithoutErrata(includesCvesWithoutErrata);
+    }, [includesCvesWithoutErrata]);
```

Comment:
There should be more dependencies here. We should at least also add `showCvesWithoutErrata`.